tot/models.py

```diff
@@ -225,3 +232,357 @@ def maybe_drop_added_dates(self, predicted, df):
+        not on a target basis.
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                dataframe containing column ``ds``, ``y``, and optionally``ID`` with all data
+
+        Returns
+        -------
+            pd.Series
+                timestamps referring to the start of the predictions.
+            np.array
+                array containing the predictions
+        """
+        # Receives df with single ID column
+        assert len(df["ID"].unique()) == 1
+
+        dates = df["ds"].iloc[self.K : -self.n_forecasts + 1].reset_index(drop=True)
+        # assemble last values based on K
+        last_k_vals_arrays = [df["y"].iloc[i : i + self.K].values for i in range(0, dates.shape[0])]
+        last_k_vals = np.stack(last_k_vals_arrays, axis=0)
+        # Compute the predictions
+        predicted = np.array([last_k_vals[:, i % self.K] for i in range(self.n_forecasts)]).T
+
+        # No un-scaling and un-normalization needed. Operations not applicable for naive model
+        return dates, predicted
+
+    def _reshape_raw_predictions_to_forecst_df(self, df_i, predicted):  # Todo outsource to df_utils?
+        """Turns forecast-origin-wise predictions into forecast-target-wise predictions.
+
+        Parameters
+        ----------
+            df : pd.DataFrame
+                input dataframe
+            predicted : np.array
+                Array containing the predictions
+
+
+        Returns
+        -------
+            pd.DataFrame
+                columns ``ds``, ``y``, optionally ``ID`` and [``yhat<i>``],
+
+                Note
+                ----
+                where yhat<i> refers to the i-step-ahead prediction for this row's datetime.
+                e.g. yhat3 is the prediction for this datetime, predicted 3 steps ago, "3 steps old".
+        """
+        cols = ["ds", "y", "ID"]  # cols to keep from df
+        fcst_df = pd.concat((df_i[cols],), axis=1)
+        # create a line for each forecast_lag
+        # 'yhat<i>' is the forecast for 'y' at 'ds' from i steps ago.
+        for forecast_lag in range(1, self.n_forecasts + 1):
+            forecast = predicted[:, forecast_lag - 1]
+            pad_before = self.K + forecast_lag - 1
+            pad_after = self.n_forecasts - forecast_lag
+            yhat = np.concatenate(
+                ([np.NaN] * pad_before, forecast, [np.NaN] * pad_after)
+            )  # add pad based on n_forecasts and current forecast_lag
+            name = f"yhat{forecast_lag}"
+            fcst_df[name] = yhat
+
+        return fcst_df
+
+
+@dataclass
+class NaiveModel(SeasonalNaiveModel):
+    """
+    A `NaiveModel` is a naive model that forecasts future values of a target series as the value of the
+    last observation of the target series. The NaiveModel is SeasonalNaiveModel with K=1.
+
+    The naive models act on time series data having ``fit()`` and ``predict()`` methods.
```

Comment:
I think you can remove most of the things below in the docstring?

Reply:
this should not receive any `K`, too?

Reply:
maybe it is simpler to reimplement it after all?

Reply:
@ourownstory Let`s quickly discuss the docstring and reimplementation verbally. :)